uk.co.compucorp.civicrm.hrleaveandabsences/xml/schema/CRM/HRLeaveAndAbsences/LeaveBalanceChange.xml

```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+
+<table>
+  <base>CRM/HRLeaveAndAbsences</base>
+  <class>LeaveBalanceChange</class>
+  <name>civicrm_hrleaveandabsences_leave_balance_change</name>
+  <comment>Store balance changes to a Leave Period Entitlement</comment>
+  <add>4.4</add>
+  <log>true</log>
+
+  <field>
+    <name>id</name>
+    <type>int unsigned</type>
+    <required>true</required>
```

Comment:
I also may be mistaken but I interrupt the required tag as a DAO rule and since BAO class extends DAO class , calling the create method without id parameter will lead it to fail since the required ID field is not present .